### PR TITLE
Moved remaining provisioners logic to core, added docstrings

### PR DESCRIPTION
**Pull request by abrown-sg**
With the commands moving to their own class, there wasn't much sense having core logic spread into the provisioners file.

Added docstrings.

Started process of updating changelog during development to better track changes. Will update version once we decide on a new release.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,5 +1,10 @@
 CHANGELOG for molecule
 ======================
+1.0.7.dev1
+----
```
Comment (retr0h) [inline CHANGELOG.rst]: Do we really need a changelog entry for this?  Seems this work can go out as 1.1.0 once  the 'static' provider is done, no?  It's all prep work for that.

Comment (rgreinho) [inline CHANGELOG.rst]: if you have ``dev1`` at the end of the version, it means there was an issue with tagging/packaging.

File: molecule/commands.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
 
 import molecule.utilities as utilities
 import molecule.validators as validators
-from provisioners import Ansible
+from molecule.core import Molecule
```
Comment (rgreinho) [inline molecule/commands.py]: Bump it 2 lines higher, 'c' > 'u'.

**Comment by abrown-sg** [comment]
Eh, I was just thinking of a way to document changes along the way. I'll axe it.

File: molecule/commands.py (hunk 2 above)
Comment (abrown-sg) [inline molecule/commands.py]: John asked me to keep froms below imports.

Comment (retr0h) [inline molecule/commands.py]: And yes I know this is holly war ground for that particular topic. :P

Comment (rgreinho) [inline molecule/commands.py]: Haha, not so much holly war though: http://docs.openstack.org/developer/hacking/#imports

**Comment by abrown-sg** [comment]
@retr0h @rgreinho So, :+1:  :-1:? :)

**Comment by rgreinho** [comment]
:+1: I'm not going to hold you for an alphabet problem, but I still think we should follow the OpenStack guidelines.

**Pull request merged**